### ISSUE TITLE
Add screenshot options

### ISSUE THREAD

**Issue by andrewvc** [elastic/synthetics#308]
`--screenshots` should take an option of: `off`, `on`, `only-on-failure` or `space-efficient`.

`space-efficient` be the default in heartbeat and would retain just the last step's screenshot on successful runs, and retain all steps on errors. This is important because screenshots, even with the efficiency gains we've recently gotten, just take up too much space and cost too much. Even a handful of journeys can takeup ~1g/day with those optimizations. This should cut down disk usage dramatically. 

Counterpart of https://github.com/elastic/beats/issues/25201

@liciavale  @drewpost @paulb-elastic @vigneshshanmugam thoughts on this approach? For the `space-efficient` option it'd mean that on the steps page in Uptime we'd usually have missing screenshots.

**Comment by paulb-elastic** [comment]
I guess if we need to drastically reduce space, then we need a drastic option like this.  I suspect having the screenshot of just the last step is probably only marginally better than no screenshots at all.

In isolation, the screenshot of the last step is probably not that useful.  Where it comes in more useful is when showing a good screenshot as a comparison with a failed test.  In this scenario, we can only do that if the error happens on the last step.

What about only storing the ‘latest’ screenshot for good tests (but for all steps).  I.e. continually overwriting the good screenshots as each test result comes in (probably still split by location, as the screenshot may be different geographically).  This means when there’s a failure, we can still show a comparison with the last good test (which is what we do in Uptime anyway).

(Incidentally https://github.com/elastic/uptime/issues/330 was raised because we don’t currently honour location when we show the comparison)

This won’t help if the test is always in error (but nor would the current suggestion for `space-efficient` either).

**Comment by liciavale** [comment]
I agree with Paul on this. Also - I think we could also probably reduce the size of the image, atm we have it full screen but do we need it that big?

**Comment by justinkambic** [comment]
In my opinion, the name `space-efficient` doesn't provide enough context into how the feature achieves the savings. If we implement this feature I think we should name it something else. Users will need to look up the meaning in the docs if they want to find out what this does. Maybe it could be `final-step` or `final-step-only`.

**Comment by vigneshshanmugam** [comment]
For now we are going only with `on|off|only-on-failure`, We will keep the discussion open for the default option that makes most sense for the users instead of the current one which is always `on`.
